File: helfomat-core/src/main/java/de/helfenkannjeder/helfomat/core/picture/PictureStorageService.java
package de.helfenkannjeder.helfomat.core.picture;

import java.io.InputStream;
import java.nio.file.Path;

/**
 * @author Valentin Zickner
 */
public interface PictureStorageService {

    PictureId savePicture(String url, PictureId pictureId) throws DownloadFailedException;

    PictureId savePicture(byte[] bytes, PictureId pictureId) throws DownloadFailedException;

    PictureId savePicture(PictureId pictureId, InputStream inputStream) throws DownloadFailedException;

    Path getPicture(PictureId pictureId);

    Path getPicture(PictureId pictureId, String size);

    boolean existPicture(PictureId pictureId);
}

File: helfomat-core/src/main/kotlin/de/helfenkannjeder/helfomat/core/picture/PictureStorageService.kt
package de.helfenkannjeder.helfomat.core.picture

import java.io.InputStream
import java.nio.file.Path

/**
 * @author Valentin Zickner
 */
interface PictureStorageService {

    fun savePicture(url: String, pictureId: PictureId): PictureId
    fun savePicture(bytes: ByteArray, pictureId: PictureId): PictureId
    fun savePicture(pictureId: PictureId, inputStream: InputStream): PictureId
    fun getPicture(pictureId: PictureId): Path
    fun getPicture(pictureId: PictureId, size: String): Path
    fun existPicture(pictureId: PictureId): Boolean

}